src/hotspot/share/prims/universalUpcallHandler.cpp

```diff
@@ -51,7 +51,8 @@ void ProgrammableUpcallHandler::attach_thread_and_do_upcall(jobject rec, address
   JNIEnv* p_env = nullptr;
   if (thread == nullptr) {
     JavaVM_ *vm = (JavaVM *)(&main_vm);
-    vm->functions->AttachCurrentThread(vm, (void**) &p_env, nullptr);
+    jint result = vm->functions->AttachCurrentThread(vm, (void**) &p_env, nullptr);
+    guarantee(result == JNI_OK, "Could not attach thread for upcall. JNI error code: %d", result);
```

Comment:
I'm assuming you don't have a mechanism for conveying an error back to the original entry point used by the native thread? Attaching an existing thread should only fail if we run out of C-Heap, so we're on the brink of aborting anyway, but still the guarantee here is not ideal.

Reply:
Yeah, we have no idea where/how to report such and error. The native code might just call a function through a function pointer like `void(*)(void)` i.e. no place to return an error code. Also, since it's a previously non-attached thread, there is no JavaFrameAnchor that gives us a last Java frame we could jump back to and throw an exception.

Is there's a more explicit way to exit with an error, other than a `guarantee` that you would prefer here?